librz/bin/pdb/tpi.c

```diff
@@ -781,165 +306,179 @@ static ut64 get_numeric_val(SNumeric *numeric) {
+	case LF_STRUCTURE_19:
+	case LF_CLASS_19: {
+		tmp = rz_bin_pdb_get_type_by_index(
+			stream, ((Tpi_LF_Structure_19 *)t->type_data)->field_list);
+		Tpi_LF_FieldList *lf_struct19 = tmp ? tmp->type_data : NULL;
 		return lf_struct19 ? lf_struct19->substructs : NULL;
 	}
-	case eLF_ENUM: {
-		tmp = rz_bin_pdb_stype_by_index(((SLF_ENUM *)t->type_info)->field_list);
-		SLF_FIELDLIST *lf_enum = tmp ? tmp->type_data.type_info : NULL;
+	case LF_ENUM: {
+		tmp = rz_bin_pdb_get_type_by_index(
+			stream, ((Tpi_LF_Enum *)t->type_data)->field_list);
+		Tpi_LF_FieldList *lf_enum = tmp ? tmp->type_data : NULL;
 		return lf_enum ? lf_enum->substructs : NULL;
 	}
 	default:
 		return NULL;
 	}
 }
 
-static char *get_type_name(void *type) {
+RZ_API char *rz_bin_pdb_get_type_name(TpiType *type) {
 	rz_return_val_if_fail(type, NULL);
-	STypeInfo *t = (STypeInfo *)type;
-	switch (t->leaf_type) {
-	case eLF_MEMBER: {
-		SLF_MEMBER *lf_member = t->type_info;
+	switch (type->leaf_type) {
+	case LF_MEMBER: {
+		Tpi_LF_Member *lf_member = type->type_data;
 		return lf_member->name.name;
 	}
-	case eLF_ONEMETHOD: {
-		SLF_ONEMETHOD *lf_onemethod = t->type_info;
+	case LF_STMEMBER: {
+		Tpi_LF_StaticMember *lf_stmember = type->type_data;
+		return lf_stmember->name.name;
+	}
+	case LF_ONEMETHOD: {
+		Tpi_LF_OneMethod *lf_onemethod = type->type_data;
 		return lf_onemethod->name.name;
 	}
-	case eLF_METHOD: {
-		SLF_METHOD *lf_method = t->type_info;
+	case LF_METHOD: {
+		Tpi_LF_Method *lf_method = type->type_data;
 		return lf_method->name.name;
 	}
-	case eLF_NESTTYPE: {
-		SLF_NESTTYPE *lf_nesttype = t->type_info;
+	case LF_NESTTYPE: {
+		Tpi_LF_NestType *lf_nesttype = type->type_data;
 		return lf_nesttype->name.name;
 	}
-	case eLF_ENUM: {
-		SLF_ENUM *lf_enum = t->type_info;
+	case LF_ENUM: {
+		Tpi_LF_Enum *lf_enum = type->type_data;
 		return lf_enum->name.name;
 	}
-	case eLF_ENUMERATE: {
-		SLF_ENUMERATE *lf_enumerate = t->type_info;
+	case LF_ENUMERATE: {
+		Tpi_LF_Enumerate *lf_enumerate = type->type_data;
 		return lf_enumerate->name.name;
 	}
-	case eLF_CLASS:
-	case eLF_STRUCTURE: {
-		SLF_STRUCTURE *lf_struct = t->type_info;
+	case LF_CLASS:
+	case LF_STRUCTURE: {
+		Tpi_LF_Structure *lf_struct = type->type_data;
 		return lf_struct->name.name;
 	}
-	case eLF_CLASS_19:
-	case eLF_STRUCTURE_19: {
-		SLF_STRUCTURE_19 *lf_struct_19 = t->type_info;
+	case LF_CLASS_19:
+	case LF_STRUCTURE_19: {
+		Tpi_LF_Structure_19 *lf_struct_19 = type->type_data;
 		return lf_struct_19->name.name;
 	}
-	case eLF_ARRAY: {
-		SLF_ARRAY *lf_array = t->type_info;
+	case LF_ARRAY: {
+		Tpi_LF_Array *lf_array = type->type_data;
 		return lf_array->name.name;
 	}
-	case eLF_UNION: {
-		SLF_UNION *lf_union = t->type_info;
+	case LF_UNION: {
+		Tpi_LF_Union *lf_union = type->type_data;
 		return lf_union->name.name;
 	}
 	default:
 		return NULL;
 	}
 }
 
-static ut64 get_type_val(void *type) {
+RZ_API ut64 rz_bin_pdb_get_type_val(TpiType *type) {
```

Comment:
And here.
Btw, do we really need these functions to the the API? cc @thestr4ng3r @ret2libc